File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int min_profit = INT_MAX;
        int max_profit = 0;
        for (int i = 0; i < prices.size(); i++){
            if(prices[i] < min_profit){
                min_profit = prices[i];
            }
            else{
                max_profit = max(max_profit, prices[i] - min_profit);
            }
        }
        return max_profit;
        
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int minProfit = Integer.MAX_VALUE;
        int maxProfit = 0;
        for(int i=0;i<prices.length;i++){
            if(prices[i]<minProfit){
                minProfit = prices[i];
            }
            else{
                maxProfit = Math.max(maxProfit,prices[i]-minProfit);
            }
        }
        return maxProfit;
    }
}